server/controllers/stock/core.js

```diff
@@ -956,39 +977,41 @@ async function getInventoryQuantityAndConsumption(params) {
 
   const clause = ` GROUP BY l.inventory_uuid, m.depot_uuid ${emptyLotToken} ORDER BY ig.name, i.text `;
 
-  let filteredRows = await getLots(sql, params, clause);
-  if (filteredRows.length === 0) { return []; }
+  const filteredRows = await getLots(sql, params, clause);
+  let _filteredRows = params.paging ? filteredRows.rows : filteredRows;
+
+  if (_filteredRows.length === 0) { return []; }
 
   const settingsql = `
     SELECT month_average_consumption, average_consumption_algo, min_delay, default_purchase_interval
     FROM stock_setting WHERE enterprise_id = ?
   `;
 
-  const opts = await db.one(settingsql, filteredRows[0].enterprise_id);
+  const opts = await db.one(settingsql, _filteredRows[0].enterprise_id);
 
   // add the minimum delay to the rows
-  filteredRows.forEach(row => {
+  _filteredRows.forEach(row => {
```

Comment:
I find using the _ prefix not very helpful for making the code easy to understand.  Rather than `_filteredRows`, I suggest using `filteredRowsPaged` or `filteredRowsPaging` (or something similar) which is more descriptive and easy to understand why `_filteredRows` is different from `filteredRows`.